core/consensus/types.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2019 ChainSafe Systems (ON) Corp.
+// This file is part of gossamer.
+//
+// The gossamer library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The gossamer library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the gossamer library. If not, see <http://www.gnu.org/licenses/>.
+
+package consensus
+
+import (
+	"math/big"
+
+	"github.com/ChainSafe/gossamer/common"
+)
+
+//type used to store an authority Id
+type authorityId [32]byte
+
+//length of vrf output
+type VRFOutput [32]byte
+
+//signature size
+type Signature [32]byte
+
+type Hash [32]byte
+
+//Justified Header
+type JustifiedHeader struct {
+	BlockHeader   common.BlockHeader
+	Justification [64]byte
+	AuthorityIds  []authorityId
+}
+
+type Block struct {
```

Comment:
looking at the spec, a block only seems to have block header + block body

slot number, vrf, signature etc are all in the babe header: see Definition 4.10
should probably create a separate type for that

Reply:
Remoed. See other comment.